tests/rados/test_bug_fixes.py

```diff
@@ -698,3 +702,110 @@ def run(ceph_cluster, **kw):
             "Verification of CEPHADM_STRAY_DAEMON warnig for all combination completed"
         )
         return 0
+
+    if config.get("test_osd_spec_update"):
+        doc = (
+            "\n# bug : https://bugzilla.redhat.com/show_bug.cgi?id=2304314"
+            "\n Verify that OSDs can be added/moved to new spec"
+            "\n\t 1. Deploy a cluster with at least 1 mon, mgr, OSD, and RGW each"
+            "\n\t 2. Remove 1 OSD daemon, note the daemon ID"
+            "\n\t 3. Add back the OSD via ceph orch daemon add command"
+            "\n\t 4. Note the OSD service created for the new OSD is unmanaged"
+            "\n\t 5. Move the OSD to the older managed service"
+        )
+
+        log.info(doc)
+        log.info("Running test to Verify that OSDs can be added/moved to new spec")
+
+        try:
+            if float(rhbuild.split("-")[0]) < 8.1:
+                log.info("Passing without execution, BZ yet to be backported")
+                return 0
+
+            target_osd = random.choice(rados_obj.get_osd_list(status="UP"))
+            host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=target_osd)
+            osd_spec_name = service_obj.get_osd_spec(osd_id=target_osd)
+            dev_path = get_device_path(host, target_osd)
+            log_msg = (
+                f"\nSelected OSD for test : \n"
+                f"osd device path  : {dev_path},\n osd_id : {target_osd},\n host.hostname : {host.hostname}\n"
+                f"OSD spec : {osd_spec_name}\n"
+            )
+            log.info(log_msg)
+            osd_spec_name = service_obj.get_osd_spec(osd_id=target_osd)
+            rados_obj.set_service_managed_type(service_type="osd", unmanaged=True)
+            utils.osd_remove(ceph_cluster, target_osd)
+            method_should_succeed(wait_for_clean_pg_sets, rados_obj)
+            method_should_succeed(
+                utils.zap_device, ceph_cluster, host.hostname, dev_path
+            )
+            method_should_succeed(
+                wait_for_device_rados, host, target_osd, action="remove"
+            )
+
+            # Adding the removed OSD back and checking the cluster status
+            utils.add_osd(ceph_cluster, host.hostname, dev_path, target_osd)
+            method_should_succeed(wait_for_device_rados, host, target_osd, action="add")
+            log.debug("Completed addition of the OSD back to cluster.")
+
+            # Checking if the newly added OSD is without placement
+            if service_obj.unmanaged_osd_service_exists():
```

Comment:
At this point aren't all the OSD Orch services in `unmanaged` state due to execution of line 736?

Reply:
this is check for a OSD orch service, that cannot be set in managed state. Basically a check for un-manage-able placement spec.